lib/view_component/instrumentation.rb

```diff
@@ -5,14 +5,16 @@
 module ViewComponent # :nodoc:
   module Instrumentation
     def self.included(mod)
-      mod.prepend(self)
+      mod.prepend(self) unless ancestors.include?(ViewComponent::Instrumentation)
```

Comment:
Nice find here!

Reply:
Thanks.  :)

I have to admit it took me a couple of hours to suss this out.  I was scratching my head looking at `caller` outputs and trying to figure out how they could be different between Ruby 3.0 and Ruby 3.1.  Very satisfying to get an answer.